client/client.go

```diff
@@ -116,7 +117,14 @@ func CreateClient(kubeconfigPath string, overrides *clientcmd.ConfigOverrides) (
 
 				token, err := oauthConfig.TokenSource(context.TODO(), &oauth2.Token{RefreshToken: refreshToken}).Token()
 				if err != nil {
-					return nil, err
+					var e *oauth2.RetrieveError
+					if errors.As(err, &e) {
+						if strings.HasSuffix(err.Error(), `{"error":"invalid_grant"}`) {
```

Comment:
Should probably have an `else` for this, as any error could be contained in the `*oauth2.RetrieveError`, but `invalid_grant` is the only case that is handled.

Reply:
Fixed.